### PR TITLE
Add parentchain-pallet

### PR DESCRIPTION
**Pull request by echevrier**
to get parentchain pallet.
Update pallet to new substrate

### CODE REVIEW

File: parentchain/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    "frame-system/std",
+    "sp-core/std",
+    "sp-io/std",
+    "sp-runtime/std",
+    "sp-std/std",
+    "log/std",
+]
+
+runtime-benchmarks = [
+    "frame-benchmarking",
```
Comment (clangenb) [inline parentchain/Cargo.toml:49]: We need frame benchmarking/std too.

Comment (clangenb) [reply parentchain/Cargo.toml:49]: Why do we have this dependency anyhow? I don't see any benchmarks.

File: parentchain/src/weights.rs [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+pub trait WeightInfo {
+	fn set_block() -> Weight;
+}
+
+/// Weights for pallet_parentchain using the Integritee parachain node and recommended hardware.
+impl WeightInfo for () {
+	fn set_block() -> Weight {
+		10_000 as Weight
+	}
+}
```
Comment (clangenb) [inline parentchain/src/weights.rs:13]: Where did you get these weights?

**Review by clangenb — COMMENTED** [review]
Looks good, I am only wondering about the benchmarking, and I don't get the prs name.

File: parentchain/Cargo.toml (hunk 1 above)
Comment (echevrier) [reply parentchain/Cargo.toml]: I removed frame-benchmarking, as no benchmarks are needed for now

File: parentchain/src/weights.rs (hunk 2 above)
Comment (echevrier) [reply parentchain/src/weights.rs:13]: This was decided by @gaudenzkessler and @brenzi

File: parentchain/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+homepage = "https://integritee.network/"
+repository = "https://github.com/integritee-network/pallet-teerex/"
+license = "Apache-2.0"
+edition = "2018"
+resolver = "2"
+
+[dependencies]
+serde = { features = ['derive'], optional = true, version = '1.0.101' }
+log = { version = "0.4.14", default-features = false }
+hex-literal = { version = "0.3.2", optional = true }
```
Comment (haerdib) [inline parentchain/Cargo.toml:15]: Can you remove that one? Doesn't seem to be used.

**Review by haerdib — APPROVED** [review]
LGTM. Only one small remark.

**Comment by echevrier** [comment]
> I don't get the prs name.

I only want to merge the updated to new substrate "teaclave-1.1.3" branch into the "master" branch

**Review by clangenb — APPROVED** [review]

**Pull request merged**